src/parse/command.rs

```diff
@@ -88,9 +88,15 @@ pub fn Mail_parameters(input: &[u8]) -> IResult<&[u8], Vec<Parameter>> {
 
 /// esmtp-param = esmtp-keyword ["=" esmtp-value]
 pub fn esmtp_param(input: &[u8]) -> IResult<&[u8], Parameter> {
-    map(
+    map_res(
         tuple((esmtp_keyword, opt(preceded(tag(b"="), esmtp_value)))),
-        |(keyword, value)| Parameter::new(keyword, value),
+        |(keyword, value)| match keyword {
+            "SIZE" => value.unwrap_or_default().parse().map(Parameter::Size),
```

Comment:
Can we match case-insensitive here?

Reply:
>  I would only make the suggestion to match case-insensitive on "SIZE". 

Actually that's exactly what I just did before reading your comment here cause it brakes my manual test. :)

> Regarding breaking the API due to exhaustiveness: Good point. But I think we can address that later when we start to care about API breakages

Sure, only thing is we need to remember. :-D IMHO it's best to handle such things before they're actually needed.

> Maybe using #[non_exhaustive] is all we need.

Yeah.

> But there are some tradeoffs we maybe want to discuss.

Given that we've a catch-all case already, I think in this case non_exhaustive makes sense but your call if we want to do that now or later.

Reply:
> Sure, only thing is we need to remember. :-D IMHO it's best to handle such things before they're actually needed.

Yeah :-) Actually... I think that it might really be a good idea to make it non_exhaustive right away. So, feel free to do so now!

Reply:
> feel free to do so now!

Done.